src/main/java/de/tum/in/www1/artemis/web/rest/LectureResource.java

```diff
@@ -231,29 +221,24 @@ else if (lectureUnit instanceof AttachmentUnit) {
     }
 
     /**
-     * DELETE /lectures/:id : delete the "id" lecture.
+     * DELETE /lectures/:lectureId : delete the "lectureId" lecture.
      *
-     * @param id the id of the lecture to delete
+     * @param lectureId the id of the lecture to delete
      * @return the ResponseEntity with status 200 (OK)
      */
-    @DeleteMapping("/lectures/{id}")
+    @DeleteMapping("/lectures/{lectureId}")// TODO: should be /courses/{courseId}/lectures/{lectureId}
     @PreAuthorize("hasRole('EDITOR')")
-    public ResponseEntity<Void> deleteLecture(@PathVariable Long id) {
+    public ResponseEntity<Void> deleteLecture(@PathVariable Long lectureId) {
         User user = userRepository.getUserWithGroupsAndAuthorities();
-        Optional<Lecture> optionalLecture = lectureRepository.findByIdWithPostsAndLectureUnitsAndLearningGoals(id);
-        if (optionalLecture.isEmpty()) {
-            return ResponseEntity.notFound().build();
-        }
-        Lecture lecture = optionalLecture.get();
-        if (!authCheckService.isAtLeastEditorInCourse(lecture.getCourse(), user)) {
```

Comment:
I think you have forgotten to add 
authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.EDITOR, lecture.getCourse(), user);
after you have deleted this part. Or is there a check that covers this?

Reply:
Look at line 233 ^^